platform/darwin/src/MGLStyleLayer.h

```diff
@@ -69,6 +69,17 @@ MGL_EXPORT
  */
 @property (nonatomic, assign) float minimumZoomLevel;
 
+#pragma mark Layer Transitions
+
+- (void)setTransition:(MGLTransition)transition forKey:(NSString *)key;
```

Comment:
I'm thinking we might want to call it `-setTransition:forAttribute:` to be more specific about the kind of key that's being transitioned.